src/Mod/PartDesign/Gui/ViewProviderSketchBased.h

```diff
@@ -23,14 +23,14 @@
 #ifndef VIEWPROVIDERSKETCHBASED_H_QKP3UG9A
 #define VIEWPROVIDERSKETCHBASED_H_QKP3UG9A
 
-#include "ViewProvider.h"
+#include "ViewProviderAddSub.h"
 
 namespace PartDesignGui {
 
 /**
  * A common base class for Sketch based view providers
  */
-class PartDesignGuiExport ViewProviderSketchBased : public ViewProvider
+class PartDesignGuiExport ViewProviderSketchBased : public ViewProviderAddSub
```

Comment:
The reason for this change is tha AddSub has some nice preview functionality that I found useful for aiding the user in creating a subractive helix.

Reply:
Is there a reason why ViewProviderHelix doesn't inherit from ViewProviderAddSub, instead?
Due to the change of the inheritance the classes ViewProviderPad, ViewProviderRevolution, ... require extra memory for nothing.

Reply:
None other than my inexperience.  Fixed